minidump-processor/src/process_state.rs

```diff
@@ -122,6 +122,87 @@ impl From<MinidumpLinuxProcStatus<'_>> for LinuxProcStatus {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum Limit {
+    Error,
+    Unlimited,
+    Limited(u64)
+}
+
+impl serde::Serialize for Limit {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match *self {
+            Limit::Error => serializer.serialize_str("err"),
+            Limit::Unlimited => serializer.serialize_str("unlimited"),
+            Limit::Limited(val) => serializer.serialize_u64(val)
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct LinuxProcLimit {
+    pub name: String,
+    pub soft: Limit,
+    pub hard: Limit,
+    pub unit: String
+}
+
+#[derive(Debug, Clone)]
+pub struct LinuxProcLimits {
+    pub limits: HashMap<String, LinuxProcLimit>
+}
+
+fn parse_limit(s: String) -> Limit {
+    match s.trim() {
+        "unlimited" => Limit::Unlimited,
+        val => Limit::Limited(val.parse::<u64>().unwrap_or(0))
+    }
+}
+
+impl From<MinidumpLinuxProcLimits<'_>> for LinuxProcLimits {
+    fn from(limits: MinidumpLinuxProcLimits) -> Self {
+        let hash: HashMap::<String, LinuxProcLimit> = limits
+            .iter()
```

Comment:
Since the first line always contains the header you could do this here:
```rust
    .iter()
    .skip(1)
    .map(...)
```
... and then remove the `m[0].contains("Max")` condition later, as all the lines will have it already

Reply:
We have to skip two lines, based on the test, the first one is empty and the second one is headers.